stake-pool/single-pool/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+#![deny(missing_docs)]
+
+//! A program for creating and managing pools of stake
+
+pub mod error;
+pub mod instruction;
+pub mod processor;
+
+#[cfg(not(feature = "no-entrypoint"))]
+pub mod entrypoint;
+
+// Export current sdk types for downstream users building with a different sdk version
+pub use solana_program;
+use solana_program::pubkey::Pubkey;
+
+// XXX change this
+solana_program::declare_id!("3cqnsMsT6LE96pxv7GR4di5rLqHDZZbR3FbeSUeRLFqY");
+
+const POOL_STAKE_PREFIX: &[u8] = b"stake";
+const POOL_AUTHORITY_PREFIX: &[u8] = b"authority";
+const POOL_MINT_PREFIX: &[u8] = b"mint";
+
+const MINT_DECIMALS: u8 = 9;
+
+const VOTE_STATE_OFFSET: usize = 4;
+
+// XXX obviously i hate this and if anyone has a better idea...
+// XXX this is only pub for testing purposes because the struct is in tests. not sure if theres a better way
+//    4 (enum)
+// + 32 (node_pubkey)
+// + 32 (authorized_voter)
+// +  8 (authorized_voter_epoch)
+// + (32 + 8 * 3) * 32 + 8 (prior_voters)
+// = 1876
+/// doc
+pub const LEGACY_VOTE_STATE_OFFSET: usize = 1876;
```

Comment:
also someone who is familiar with `VoteStateVersions` should sanity check this, im assuming that `VoteState0_23_5` (and `VoteState` but thats more obvious) onchain is _always_ wrapped in the enum, ie that the four byte offset is _always_ there. in other words, that `VoteStateVersions` was planned from there very beginning as future-proofing, before there were multiple versions

the reason i assume this has to be the case is `VoteState::deserialize` is written to parse both versions and upconvert the old one, and there is no way it could work if the enum tag wasnt in the account data for the old version